controllers/awsmachine_controller.go

```diff
@@ -673,6 +673,50 @@ func (r *AWSMachineReconciler) reconcileOperationalState(ec2svc services.EC2Inte
 		return err
 	}
 
+	// check if the remote kubeconfig works and annotate the cluster
+	if _, ok := machineScope.InfraCluster.InfraCluster().GetAnnotations()[scope.KubeconfigReadyAnnotation]; !ok && machineScope.IsControlPlane() {
```

Comment:
I'm not sure about this as the mechanism to decide if we carry on with the OIDC reconciliation.

Reply:
Added a suggestion in the awscluster_controller.go